crates/subspace-service/src/config.rs

```diff
@@ -183,10 +183,11 @@ impl From<SubstrateConfiguration> for Configuration {
                 }),
             telemetry_endpoints: configuration.telemetry_endpoints,
             default_heap_pages: None,
-            // Offchain worker is not used in Subspace
+            // Offchain worker is not used and indexing is used by MMR gadget to
+            // prune fork mmr leaves once a given block is finalized
             offchain_worker: OffchainWorkerConfig {
                 enabled: false,
-                indexing_enabled: false,
+                indexing_enabled: true,
```

Comment:
Read further commits, is this just for RPC?

Reply:
> If offchain indexing is not enabled then there is nothing to prune

That might not be true actually. MMR pallet uses aux storage through offchain storage api to store MMR leaf during initialization. This stored in a fork aware key.

`MMR gadget` should be enabled so that non-canonical MMR and their leaf nodes are pruned once a given block is finalized so that there is only canonical mmr and their leafs. 

Honestly, I was a bit in the middle ground when I went with this `indexing_enabled` flag. 
Reason I put this behind the `indexing_enabled` is since polkadot uses it and I did not not want to break anything underneath with this flag not enabled but I think we can get rid of this and just run the MMR gadget on every node since all the Gadget seems to be doing is to prune the non canionical leafs at that block height.

> Read further commits, is this just for RPC?

I have added RPC in the last as a usability feature where someone can actually use the rpc through a side-car to generate and verify proofs. And no strong opinion on whether to keep it or not but I think it is okey to keep it. For XDM, we can just use runtime_apis.

Reply:
> If this flag set to false the offchain storage changes (i.e. mmr leaf) may just drop silently? I'm sure though.

That was my expectation, hence the comment

Reply:
If that is the case it may lead to hard fork between nodes with `indexing_enabled` set to `true` and those set to `false` since the `pallet-mmr` will compute and store different mmr root.

Reply:
It will not. Offchain indexing is for offchain purposes. Unless we have host function that calls into offchain storage (which we really shouldn't have), this is a one-way data transfer from runtime into offchain storage. Whether it is enabled or disabled makes no difference for runtime logic.

Reply:
Yeah, took a closer look, only runtime states are used when computing the mmr root. But when generating the proof and verifying with `verify_proof` runtime API the offchain storage is used (`verify_proof_stateless` doesn't use offchain storage though), it will lead to different results depending on the `indexing_enabled` value.

Reply:
>  If offchain indexing is not enabled then there is nothing to prune, so comment above doesn't make a lot of sense to me. And if offchain worker is not enabled, there will be no way to read indexed data anyway IIUC.

Looked into this and its works out with indexing enabled and I see the discussion above lead to that 👍🏼